src/server/scripts/BrokenIsles/Dalaran/VioletHold/instance_violethold.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AreaTriggerAI.h"
+#include "ScriptMgr.h"
+#include "Creature.h"
+#include "CreatureGroups.h"
+#include "GameObject.h"
+#include "InstanceScript.h"
+#include "Map.h"
+#include "violethold.h"
+
+#define MAX_ENCOUNTER 8
+
+/* Stonecore encounters:
+0 - Festerface               0% Done
+1 - Shivermaw                0% Done
+2 - Blood-Princess Thal'ena  0% Done
+3 - Mindflayer Haahrj        0% Done
+4 - Millificent Manastorm    0% Done
+5 - Anub'esset               0% Done
+6 - Sael'orn                 0% Done
+7 - Fel Lord Betrug          0% Done
+*/
+
+// TO-DO:
+// - Find Any and All.
+
+class instance_violethold : public InstanceMapScript
+{
+    public:
+        instance_violethold() : InstanceMapScript(SCScriptName, 1544) { }
+
+        struct instance_violethold_InstanceScript : public InstanceScript
+        {
+            instance_violethold_InstanceScript(InstanceMap* map) : InstanceScript(map) { }
+
+            void OnPlayerEnter(Player* player) override
+            {
+                return InstanceScript::OnPlayerEnter(player);
+            }
+
+            void OnGameObjectCreate(GameObject* go) override
+            {
+                return InstanceScript::OnGameObjectCreate(go);
+            }
+
+            void OnCreatureCreate(Creature* creature) override
+            {
+                return InstanceScript::OnCreatureCreate(creature);
+            }
+
+            bool SetBossState(uint32 type, EncounterState state) override
+            {
+                return InstanceScript::SetBossState(type, state);
+            }
+
+        private:
+            //
+        };
+
+        InstanceScript* GetInstanceScript(InstanceMap* map) const override
+        {
+            return new instance_violethold_InstanceScript(map);
+        }
+};
+
+struct areatrigger_violet_hold_entrance_conversation : AreaTriggerAI
```

Comment:
Move to a separate script file, where also trash scripts would be put. `instance_` scripts should only contain the instance script. Create a cpp file with a name such as `violet_hold_legion.cpp` or `violet_hold_broken_isles` (without `instance_` prefix) and move this there

Reply:
I have that already done, but was going to sleep. Will push these changes today :)

Reply:
Yeah, you can move it to [src/server/scripts/BrokenIsles/Dalaran/VioletHold/violethold.cpp](https://github.com/TrinityCore/TrinityCore/pull/28460/files#diff-90c09ea0775437c87740cce0e62da51c387f7d4d6324008fb07cb727d35d3102).